tests/func/test_add.py

```diff
@@ -87,6 +90,30 @@ def test(self):
         ret = main(["add", "--recursive", self.DATA_DIR])
         self.assertEqual(ret, 0)
 
+    def test_warn_about_large_directories(self):
+        warning = (
+            "You are adding a large directory recursively,"
```

Comment:
Let's define this config fmt somewhere and use it both in the code itself and in the test. No real need to copypaste it.

Reply:
@efiop , I think there's no harm in copy pasting constants on the test, at least in this case. :slightly_smiling_face: 

If we are going to take this approach for every log message that we are testing, we would end with a lot of constants across the code base.

This is different from having a module with constants like the one used for localization/internationalization purposes (`strings.xml` - android, `locale/en.yml` - rails, etc.).

Reply:
@mroutis But at least you would only have to modify that message in one place. There is no harm in using constants, actually, it is strongly prefered over copypasting.

Reply:
Agree, @efiop , it is better to use constants than copy pasting _to reuse code_, but not to _reuse code in tests_.

It is kind of like a _purist_ perspective, I guess, that you are modifying the interface of on object by adding another attribute but just for the sake of testing, otherwise it wouldn't be needed.

I will change it on my patch, but just wanted to share my perspective :slightly_smiling_face:

Reply:
@mroutis Thanks! :slightly_smiling_face:  I agree that sometimes generalization might be too much, but in this case, it is a fairly static big message, so copypasting it would simply bloat our code. Especially since the point of that test is to simply make sure that the message is printed, not to verify the contents of it.

Reply:
@efiop , if we add the file name to the warning `"You are adding a large directory '{target}' recursively,"` is not static anymore; we would have something like:

```python
# dvc/repo/add.py
WARNING_ADD_LARGE_DIR_RECURSIVELY = (
    "You are adding a large directory '{target}' recursively,"
    " consider tracking it as a whole instead.\n"
    "{purple}HINT:{nc} Remove the generated stage files and then"
    " run {cyan}dvc add {target}{nc}"
)

# ...

        logger.warning(
            WARNING_ADD_LARGE_DIR_RECURSIVELY.format(
                purple=colorama.Fore.MAGENTA,
                cyan=colorama.Fore.CYAN,
                nc=colorama.Style.RESET_ALL,
                target=target,
            )
        )
```

can we skip it this time :stuck_out_tongue: ?

Reply:
Indeed, when `target` is interlaced with colours it is not fun to bother around with partial string formatting. Deal :slightly_smiling_face: